pkg/executor/executortype/poolmgr/gp_deployment.go

```diff
@@ -34,7 +35,18 @@ import (
 // getPoolName returns a unique name of an environment
 func (gp *GenericPool) getPoolName(env *fv1.Environment) string {
 	// TODO: get rid of resource version here
-	return strings.ToLower(fmt.Sprintf("poolmgr-%v-%v-%v", env.ObjectMeta.Name, env.ObjectMeta.Namespace, env.ObjectMeta.ResourceVersion))
+	var envPodName string
+
+	//To fit the 63 character limit
+	if len(env.ObjectMeta.Name)+len(env.ObjectMeta.Namespace) < 37 {
+		envPodName = env.ObjectMeta.Name + "-" + env.ObjectMeta.Namespace
+	} else {
+		nameLength := int(math.Min(float64(len(env.ObjectMeta.Name)), 18))
```

Comment:
Let's have a small inline function for min that deals with integers. We should avoid floating operations.
https://go.dev/play/p/18QW5-JWdl_c